### PR TITLE
Pin sphinx<7.2.0 to avoid doc build failure.

### LINKED ISSUE #25541: Sphinx job failing in CI due to duplication of `sympy.sets.sets`
All Sphinx jobs are failing with the error message `/home/runner/work/sympy/sympy/doc/src/modules/sets.rst:29: WARNING: duplicate object description of sympy.sets.sets, other instance in modules/sets, use :no-index: for one of them`

### PR DESCRIPTION
**Pull request by moorepants**
#### References to other Issues or PRs

Temporarily fixes https://github.com/sympy/sympy/issues/25541

#### Brief description of what is fixed or changed

Sphinx 7.2.0 came out today and our documentation build (`make html`) fails with this version. This fix should let CI pass until we get it working with version 7.2.0.

#### Release Notes

NO ENTRY

### CODE REVIEW

File: doc/src/explanation/gotchas.rst [hunk 1]
```diff
@@ -806,4 +806,4 @@ These will give you the function parameters and docstring for
 
 .. module:: sympy.simplify.simplify
 .. autofunction:: powsimp
-   :noindex:
+   :no-index:
```
Comment (moorepants) [inline doc/src/explanation/gotchas.rst:809]: Is this possible it will fail on Sphinx < 7.2.0?

Comment (oscarbenjamin) [reply doc/src/explanation/gotchas.rst:809]: Yes, I have just confirmed that locally.

Comment (oscarbenjamin) [reply doc/src/explanation/gotchas.rst]: I've removed that commit

**Comment by oscarbenjamin** [comment]
Set to automerge. I'll open a followup PR to unpin the version.

**Pull request merged**